### PR TITLE
Connect rails with connect_to_raillike again

### LINKED ISSUE #2479: connect_to_raillike group doesn't work as it should
If you test this patch: https://github.com/minetest/minetest_game/pull/435
you can see that when a regular trail of gunpowder starts to burn the facedir is reset. I've added param2 to the burn function but this still doesn't fix it. Any suggestions?

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #2479

Tested on Windows XP: Nodes with the same connect_to_raillike group connect to each other.
EDIT: I made the codes shorter, due a request from kwolekr

### CODE REVIEW

File: src/content_mapblock.cpp [hunk 1]
```diff
@@ -1398,97 +1398,40 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
...
-			bool is_rail_x_all[] = {false, false};
-			bool is_rail_z_all[] = {false, false};
-			is_rail_x_all[0]=is_rail_x[0] || is_rail_x_minus_y[0] || is_rail_x_plus_y[0];
-			is_rail_x_all[1]=is_rail_x[1] || is_rail_x_minus_y[1] || is_rail_x_plus_y[1];
-			is_rail_z_all[0]=is_rail_z[0] || is_rail_z_minus_y[0] || is_rail_z_plus_y[0];
-			is_rail_z_all[1]=is_rail_z[1] || is_rail_z_minus_y[1] || is_rail_z_plus_y[1];
+			u8 index = 0;
+			for (s16 y0 = -1; y0 <= 1; y0++) {
+				for (s16 xz = -1; xz <= 1; xz++) {
+					if (xz == 0 && y0 == 0)
```
Comment (Zeno-) [inline src/content_mapblock.cpp]: There is a very simple way to make this condition check and the continue unrequired...

**Comment by SmallJoker** [comment]
Are there any problems with the coding style?
I've tested those codes and haven't found any bug.

**Comment by proller** [comment]
better to store rail configuration to param2 on place and dont calculate for every mesh.

**Comment by SmallJoker** [comment]
This would require to update the nodes around the placed one, if those are rails and that's a too complicated task for me.

**Comment by est31** [comment]
param2 is also serverside, while connecting is clientside, isn't it?

**Comment by SmallJoker** [comment]
screenshot_rail_fix_before_after

**Comment by Zeno-** [comment]
709f4a5

**Pull request closed without merging**